opentelemetry-sdk/tests/test_util.py

```diff
@@ -0,0 +1,162 @@
+
+        # try to append more items
+        for idx in range(8):
+            # should drop the element without raising exceptions
+            blist.append(idx)
+
+        self.assertEqual(len(blist), list_len)
+        self.assertEqual(blist.dropped, 8)
+
+    def test_extend_no_drop(self):
+        # create empty list
+        list_len = len(TestBoundedList.base)
+        blist = BoundedList(list_len)
+        self.assertEqual(len(blist), 0)
+
+        # fill list
+        blist.extend(TestBoundedList.base)
+
+        self.assertEqual(len(blist), list_len)
+        self.assertEqual(blist.dropped, 0)
+
+        for idx in range(list_len):
+            self.assertEqual(blist[idx], TestBoundedList.base[idx])
+
+        # test __iter__ in BoundedList
+        idx = 0
+        for val in blist:
+            self.assertEqual(val, TestBoundedList.base[idx])
+            idx += 1
+
+    def test_extend_drop(self):
+        list_len = len(TestBoundedList.base)
+        # create full BoundedList
+        blist = BoundedList.from_seq(list_len, TestBoundedList.base)
+        other_list = [13, 37, 51, 91]
+
+        # try to extend with more elements
+        blist.extend(other_list)
+
+        self.assertEqual(len(blist), list_len)
+        self.assertEqual(blist.dropped, len(other_list))
+
+
+class TestBoundedDict(unittest.TestCase):
+    base = collections.OrderedDict(
+        [
+            ("name", "Firulais"),
+            ("age", 7),
+            ("weight", 13),
+            ("vaccinated", True),
+        ]
+    )
+
+    def test_negative_maxlen(self):
+        with self.assertRaises(ValueError):
+            BoundedDict(-1)
+
+    def test_from_map(self):
+        dic_len = len(TestBoundedDict.base)
+        bdict = BoundedDict.from_map(dic_len, TestBoundedDict.base)
+
+        self.assertEqual(len(bdict), dic_len)
+
+        for key in TestBoundedDict.base:
+            self.assertEqual(bdict[key], TestBoundedDict.base[key])
+
+        # map too big
+        with self.assertRaises(ValueError):
+            BoundedDict.from_map(dic_len / 2, TestBoundedDict.base)
+
+    def test_bounded_dict(self):
+        # create empty dict
+        dic_len = len(TestBoundedDict.base)
+        bdict = BoundedDict(dic_len)
+        self.assertEqual(len(bdict), 0)
+
+        # fill list
+        for key in TestBoundedDict.base:
+            bdict[key] = TestBoundedDict.base[key]
+
+        self.assertEqual(len(bdict), dic_len)
+        self.assertEqual(bdict.dropped, 0)
+
+        for key in TestBoundedDict.base:
+            self.assertEqual(bdict[key], TestBoundedDict.base[key])
+
+        # test __iter__ in BoundedList
+        for key in bdict:
+            self.assertEqual(bdict[key], TestBoundedDict.base[key])
+
+        # try to append more elements
+        for key in TestBoundedDict.base:
+            bdict["new-key" + key] = TestBoundedDict.base[key]
+
+        self.assertEqual(len(bdict), dic_len)
+        self.assertEqual(bdict.dropped, dic_len)
```

Comment:
Please also assert that the new key is in the bounded dict (i.e., old elements are dropped instead of new ones).
Same for BoundedList.

Reply:
Good one, actually I was thinking that the new elements were dropped.

Reply:
could this behavior be documented in the docstring as well?